src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -477,6 +477,11 @@ namespace ngraph
             template <>
             void CPU_Emitter::EMITTER_DECL(ngraph::op::BatchNormRelu)
```

Comment:
Wouldn't it be better to check for this condition in the node constructor since you'd catch the error closer to its origin?

Reply:
@diyessi If I understand @jbobba 's master plan, we will have multiple libraries (e.g. `libxsmm`, `mkldnn`) that might handle different cases and one or the other might be able to handle a case where batchnorm's input rank isn't necessarily 4. The checks in `BatchNormRelu::BatchNormRelu` and `CPUFusion` would then be the superset of the checks for individual libraries.

Reply:
But you still wouldn't want to create the special op if the library it needed wasn't available; so the test in the constructor would check that the required lib existed. Presumably the optimizer that introduces the fused op doesn't happen if the lib isn't available.

Reply:
CPUfusion should have a check to not generate a Batchnorm+Relu unless its of shape supported by the backend at that moment.

Reply:
cpu_emitter check is more of a paranoia check.